core/domain/translation_services.py

```diff
@@ -125,41 +125,98 @@ def add_new_translation(
     model.put()
 
 
+def _apply_changes(
+    entity_translation: translation_domain.EntityTranslation,
+    translation_changes: List[exp_domain.ExplorationChange]
+) -> None:
+    """Applies the changes to the entity_translations object.
+
+    Args:
+        entity_translation: EntityTranslation. The entity translation object.
+        translation_changes: list(ExplorationChange). The list of changes to be
+            applied.
+
+    Raises:
+        Exception. Invalid translation change cmd.
+    """
+    for change in translation_changes:
+        if change.cmd == exp_domain.CMD_EDIT_TRANSLATION:
+            # Here we use cast because we are narrowing down the type of
+            # 'change' from exp_domain.ExplorationChange to specific type of
+            # change EditTranslationsChangesCmd.
+            change = cast(
+                exp_domain.EditTranslationsChangesCmd,
+                change
+            )
+
+            if entity_translation.language_code != change.language_code:
```

Comment:
Should we validate all the changes upfront and throw an error if any of them have the incorrect language code? It seems a bit unexpected for some of the changes to silently fail.

Reply:
We only validate the schema of change list and internal values are validated inside `validation` method, in this case it will be validated after the entityTranslation.validate method is called. (This is similar to writtenTranslation and voiceover filed of exploration state.)

Reply:
I'm fine adding the validation here if you want it to be added here!

Reply:
Yes, that would be great if you don't mind. The method contract would be more explicit and easier to reason about in the future if we had to debug this.